orderer/common/cluster/comm.go

```diff
@@ -640,6 +697,76 @@ func (stream *Stream) operateWithTimeout(invoke StreamOperation, report func(err
 	}
 }
 
+func (stream *Stream) AuthenticateStream(version uint32, fromId uint64, toId uint64, signer identity.SignerSerializer) error {
+	if signer == nil {
+		return errors.New("Signer is nil")
+	}
+
+	timestamp, err := ptypes.TimestampProto(time.Now().UTC())
+	if err != nil {
+		return errors.Wrap(err, "Failed to read timestamp")
+	}
+
+	payload := &orderer.NodeAuthRequest{
+		Version:   version,
+		Timestamp: timestamp,
+		FromId:    fromId,
+		ToId:      toId,
+		Channel:   stream.Channel,
+	}
+
+	bindingFieldsHash := GetSessionBindingHash(payload)
+
+	tlsBinding, err := GetTLSSessionBinding(stream.Cluster_StepClient, bindingFieldsHash)
+	if err != nil {
+		return errors.Wrap(err, "TLSBinding failed")
+	}
+	payload.SessionBinding = tlsBinding
+
+	asnSignFields, _ := asn1.Marshal(struct {
```

Comment:
why not define the struct outside the function?

Reply:
Updated now with new struct type `type AuthRequestSignature struct {`